zkvm/src/types.rs

```diff
@@ -226,10 +226,28 @@ impl Into<Scalar> for ScalarWitness {
 }
 
 impl Predicate {
-    pub fn to_point(&self) -> CompressedRistretto {
-        match self {
-            Predicate::Opaque(point) => *point,
-            Predicate::Witness(witness) => witness.to_point(),
+    pub fn new(point: CompressedRistretto) -> Self {
```

Comment:
maybe call it `::opaque` to avoid confusion with it being a "default" initializer. It's strictly less powerful than `from_witness` and the code that uses it belongs to a verifier-only path.

```suggestion
    pub fn opaque(point: CompressedRistretto) -> Self {
```